src/pal/src/thread/process.cpp

```diff
@@ -3286,13 +3294,18 @@ getFileName(
         *lpEnd = 0x0000;
 
         /* Convert to ASCII */
-        if (!WideCharToMultiByte(CP_ACP, 0, lpCommandLine, -1,
-                                 lpFileName, MAX_LONGPATH, NULL, NULL))
+        int size = 0;
+        int length = (PAL_wcslen(lpCommandLine)+1) * sizeof(WCHAR);
+        lpFileName = lpFileNamePS.OpenStringBuffer(length);
+        if (!(size = WideCharToMultiByte(CP_ACP, 0, lpCommandLine, -1,
+                                 lpFileName, length, NULL, NULL)))
         {
+            lpFileNamePS.CloseBuffer(size);
```

Comment:
It seems we don't need to close the buffer in the error path - nothing will use it anymore and the stack string destructor doesn't care.